src/main/java/name/webdizz/sonar/grammar/spellcheck/GrammarDictionaryLoader.java

```diff
@@ -42,4 +45,20 @@ public SpellDictionary loadMainDictionary() {
             locker.unlock();
         }
     }
+
+    public Optional<SpellDictionaryHashMap> loadAlternateDictionary()  {
+        String alternateDict = settings.getString(PluginParameter.ALTERNATIVE_DICTIONARY_PROPERTY_KEY);
+        Reader reader;
+        if (alternateDict != null) {
+            alternateDict = alternateDict.replace(PluginParameter.SEPARATOR_CHAR, "\n");
+            reader = new StringReader(alternateDict);
+            try {
+                return Optional.of(new SpellDictionaryHashMap(reader));
+            } catch (IOException e) {
+                throw new UnableToLoadDictionary("There is no file with dictionary.", e);

```

Comment:
its right, but it throws by SpellDictionaryHashMap class. And now I hide it behind UnableToLoadDictionary exception